lib/rules/curly.js

```diff
@@ -56,7 +74,13 @@ module.exports = {
         var multiOnly = (context.options[0] === "multi");
         var multiLine = (context.options[0] === "multi-line");
         var multiOrNest = (context.options[0] === "multi-or-nest");
-        var consistent = (context.options[1] === "consistent");
+        var consistent = (context.options[1] === "consistent") || !!(context.options[1] && context.options[1].consistent);
+        var allowExtra = !!(context.options[1] && context.options[1].allowExtra);
+
+        // if (context.options[1] && typeof context.options[1] === 'object') {
```

Comment:
I don't think so. This is to support both object and array configuration. Currently, the rule only supports array configuration.